sdks/python/apache_beam/examples/online_clustering.py

```diff
@@ -1,20 +1,65 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import datetime
 from operator import itemgetter
-from typing import Optional
 
+import joblib
 import numpy as np
 import pandas as pd
 from sklearn.cluster import MiniBatchKMeans
 
 import apache_beam as beam
-from apache_beam import pvalue
-
-from apache_beam.coders import PickleCoder, VarIntCoder
-from apache_beam.ml.inference.base import PredictionResult
+from apache_beam.coders import PickleCoder
+from apache_beam.coders import VarIntCoder
+from apache_beam.io.filesystems import FileSystems
+from apache_beam.ml.inference.base import RunInference
+from apache_beam.ml.inference.sklearn_inference import ModelFileType
+from apache_beam.ml.inference.sklearn_inference import SklearnModelHandlerNumpy
 from apache_beam.transforms import core
 from apache_beam.transforms import ptransform
 from apache_beam.transforms.userstate import ReadModifyWriteStateSpec
 
 
+class SaveModel(beam.DoFn):
+  """Saves trained clustering model to persistent storage"""
+  def __init__(self, checkpoints_path: str):
+    self.checkpoints_path = checkpoints_path
+
+  def process(self, model):
+    # generate ISO 8601
+    iso_timestamp = datetime.datetime.utcnow().strftime("%Y%m%dT%H%M%SZ")
+    checkpoint_name = f'{self.checkpoints_path}/{iso_timestamp}.checkpoint'
+    latest_checkpoint = f'{self.checkpoints_path}/latest.checkpoint'
+    # rename previous checkpoint
+    if FileSystems.exists(latest_checkpoint):
+      FileSystems.rename([latest_checkpoint], [checkpoint_name])
+    file = FileSystems.create(latest_checkpoint, 'wb')
+    if not joblib:
+      raise ImportError(
+          'Could not import joblib in this execution environment. '
+          'For help with managing dependencies on Python workers.'
+          'see https://beam.apache.org/documentation/sdks/python-pipeline-dependencies/'  # pylint: disable=line-too-long
+      )
+
+    joblib.dump(model, file)
+
+    yield checkpoint_name
```

Comment:
I think we should yield `(checkpoint_name, model)` to give users the ability to use the model directly if they want to.